File: qh3server/qh3serverplugin_ts_proj/src/helpers/sdktypes.ts
export const LOG_LEVEL = 2
export const LOG_LEVEL_0 = 0
export const LOG_LEVEL_1 = 1
export const LOG_LEVEL_2 = 2
export const LOG_LEVEL_3 = 3
export const LOG_LEVEL_4 = 4
export const LOG_LEVEL_5 = 5

export const EXIT_SUCCESS = 0
export const EXIT_FAILURE = 1

export function debug_print(level: number, logtag: string, message: string, ...args: any[]) {
    if (level > LOG_LEVEL) {
		return;
	}
    console.log(`[${logtag}] - ${message}`, ...args);
}

export function debug_error(logtag: string, message: string, ...args: any[]) {
    console.error(`[ERROR] [${logtag}] - ${message}`, ...args);
}

export function debug_warn(logtag: string, message: string, ...args: any[]) {
    console.error(`[WARN] [${logtag}] - ${message}`, ...args);
}

File: serverplugin_lua_proj/src/helpers/sdktypes.lua


local sdktypes = {}

-- Log levels
sdktypes.LOG_LEVEL = 2
sdktypes.LOG_LEVEL_0 = 0
sdktypes.LOG_LEVEL_1 = 1
sdktypes.LOG_LEVEL_2 = 2
sdktypes.LOG_LEVEL_3 = 3
sdktypes.LOG_LEVEL_4 = 4
sdktypes.LOG_LEVEL_5 = 5
sdktypes.LOG_LEVEL_6 = 6

-- Exit codes
sdktypes.EXIT_SUCCESS = 0
sdktypes.EXIT_FAILURE = 1

-- Debugging functions
function sdktypes.debug_print(level, logtag, message, ...)
    if level > sdktypes.LOG_LEVEL then
        return
    end
    print(string.format("[%s] - %s", logtag, message), ...)
end

function sdktypes.debug_error(logtag, message, ...)
    print(string.format("[ERROR] [%s] - %s", logtag, message), ...)
end

function sdktypes.debug_warn(logtag, message, ...)
    print(string.format("[WARN] [%s] - %s", logtag, message), ...)
end

return sdktypes;